File: src/main/java/core/gui/comp/renderer/BooleanTableCellRenderer.java
package core.gui.comp.renderer;

import core.gui.comp.entry.ColorLabelEntry;
import core.gui.comp.entry.IHOTableEntry;

import java.awt.*;
import javax.swing.*;
import javax.swing.table.TableCellRenderer;


public class BooleanTableCellRenderer extends JCheckBox implements TableCellRenderer {

	public BooleanTableCellRenderer() {
		this.setOpaque(true);
		this.setBackground(Color.RED);
	}

	public Component getTableCellRendererComponent(JTable table, Object value,
			boolean isSelected, boolean hasFocus, int row, int column) {

		boolean boolValue = (Boolean) value;
		setSelected(boolValue);

		System.out.println(row + " , " + column + ":  " +  value.getClass().toString() + " " + value + "" + this.getBackground());
		System.out.println("-----------------------------------");

		setBackground(Color.RED);
		setHorizontalAlignment(SwingConstants.CENTER);
		return this;
	}

}

File: src/main/java/core/gui/comp/renderer/BooleanTableCellRenderer.kt
package core.gui.comp.renderer

import core.gui.theme.HOColorName
import core.gui.theme.ThemeManager
import java.awt.Component
import javax.swing.JCheckBox
import javax.swing.JTable
import javax.swing.table.TableCellRenderer

class BooleanTableCellRenderer : JCheckBox(), TableCellRenderer {
    init {
        setOpaque(true)
        setBackground(ThemeManager.getColor(HOColorName.TABLEENTRY_BG))
        setHorizontalAlignment(CENTER)
    }

    override fun getTableCellRendererComponent(
        table: JTable, value: Any,
        isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int
    ): Component {
        val boolValue = value as Boolean
        setSelected(boolValue)
        return this
    }
}
